README.md
# Resistor-codec-app
a resistor color coder decoder android app


changelog.txt
v1.01
-added LED torch
-added easter egg
v1.02
-fixed the 'gold-bug'
-fixed the 123'beeg'bug


src/com/example/resistorcolorcodec/rcc.java


package com.example.resistorcolorcodec;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.hardware.Camera;
import android.hardware.Camera.Parameters;
import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.os.Bundle;
import java.util.*;

import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.resistorcolorcodec.ShakeDetector.OnShakeListener;

public class rcc extends Activity {

	long counter,c,tol;
	int a,b,qux=1;
	Button black,brown,red,orange,yellow,green,blue,violet,grey,white,gold,silver,calc,r;
	Button b1,b2,m,t;
	TextView disp,d2,d3;	 
	boolean torch_engaged;
		private SensorManager mSensorManager;
	    private Sensor mAccelerometer;
	    private ShakeDetector mShakeDetector;
	    private Camera camera;
	     
	
	@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.rcc);
        counter=0;
        a=0;
        b=0;
        c=0;
        
  tol=0;  
  disp=(TextView) findViewById(R.id.textView1);
  d2=(TextView) findViewById(R.id.textView2);
  d3=(TextView) findViewById(R.id.textView3);
  
   b1=(Button) findViewById(R.id.button14);
   b2=(Button) findViewById(R.id.button15);
   m=(Button) findViewById(R.id.button16);
   t=(Button) findViewById(R.id.button17);
   r=(Button) findViewById(R.id.button18);
   black=(Button) findViewById(R.id.button1);
   brown=(Button) findViewById(R.id.button2);
   red=(Button) findViewById(R.id.button3);
   orange=(Button) findViewById(R.id.button4); 
   yellow=(Button) findViewById(R.id.button5);
   green=(Button) findViewById(R.id.button6);
   blue=(Button) findViewById(R.id.button7);
   violet=(Button) findViewById(R.id.button8);
   grey=(Button) findViewById(R.id.button9);
   white=(Button) findViewById(R.id.button10);
   gold=(Button) findViewById(R.id.button11);
   silver=(Button) findViewById(R.id.button12);
   calc=(Button) findViewById(R.id.button13);
   
   
   
   black.setOnClickListener(new View.OnClickListener() {
	
	@Override
	public void onClick(View arg0) {
		// TODO Auto-generated method stub
		
		cfunc(R.drawable.black,0,1);
		
	}
});
   
   
   brown.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			cfunc(R.drawable.brown,1,10);
			
			
		}
	});
   
   red.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.red,2,100);
			
				}
	});
   
   orange.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.orange,3,1000);
					
		}
	});
   
   yellow.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.yellow,4,10000);
			
			
		}
	});
   
   green.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			
			cfunc(R.drawable.green,5,100000);
			
			
		}
	});
  
   blue.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.blue,6,1000000);
			
				
		}
	});
  
   violet.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.violet,7,10000000);
			
				}
	});
   
   grey.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.grey,8,100000000);
			
		}
	});
   
   white.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			cfunc(R.drawable.white,9,1000000000);
			
		}
	});
   
   gold.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
		//	
			if(counter==2)
			{
			c=0;
			counter++;
			m.setBackgroundResource(R.drawable.gold);
			}
			else if(counter==3)
			{
			//
			 tol=5;	
			 t.setBackgroundResource(R.drawable.gold);	
			}
			}
	});
  
   silver.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			tol=10;
			t.setBackgroundResource(R.drawable.silver);
		}
	});
   
   calc.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
			// TODO Auto-generated method stub
			
			
			long x;
		if(1000000>c&&c>=1000)
		{	
			c=(c/1000);
			x=(((a*10)+b))*c;
			disp.setText(""+x);
		    d2.setText("KΩ±"+tol);
		    d3.setText("%");
		    counter=0;
			
		}
		else if(c>=1000000)
		{
				c=(c/1000000);
				x=(((a*10)+b))*c;
				disp.setText(""+x);
			    d2.setText("MΩ±"+tol);
			    d3.setText("%");
			    counter=0;
				
		}
		else if(c==0)
		{
			disp.setText(""+a+"."+b);
		    d2.setText("Ω±"+tol);
		    d3.setText("%");
		    counter=0;
		    
		}
		else
		{


			x=(((a*10)+b))*c;
			disp.setText(""+x);
		    d2.setText("Ω±"+tol);
		    d3.setText("% ");
		    counter=0;
			}
		    
		}
		});
  

// ShakeDetector initialization
   mSensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
   mAccelerometer = mSensorManager
           .getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
   mShakeDetector = new ShakeDetector();
   mShakeDetector.setOnShakeListener(new OnShakeListener() {

       @Override
       public void onShake(int count) {
           /*
            * The following method setup whatever you want done once the
            * device has been shook.
            */
    	   reset();
       }
   });
   
 
   
  r.setOnClickListener(new View.OnClickListener() {
	
	@Override
	public void onClick(View arg0) {
		// TODO Auto-generated method stub
		
		reset();
	
	}
});
  
	
	
 }
//func definition
	
	public void cfunc(int s,int z,int mul)
	{
		if(counter==0)
		{
		a=z;
		counter++;
		b1.setBackgroundResource(s);
		}
		else if(counter==1)
		{
			b=z;
			counter++;
			b2.setBackgroundResource(s);
		}
		else
		{
			c=mul;
			counter++;
			m.setBackgroundResource(s);
		}
	}
	
	//reset defn
	public void reset()
	{
		a=b=0;
		c=tol=counter=0;
		b1.setBackgroundResource(R.drawable.blank);
		b2.setBackgroundResource(R.drawable.blank);
		m.setBackgroundResource(R.drawable.blank);
		t.setBackgroundResource(R.drawable.blank);
		disp.setText("");
		d2.setText("");
		d3.setText(" ");	
	
	}
	
	
	
	
	@Override
	    public void onResume() {
	        super.onResume();
	        // Add the following line to register the Session Manager Listener onResume
	        mSensorManager.registerListener(mShakeDetector, mAccelerometer,    SensorManager.SENSOR_DELAY_UI);
	    }
	 
	    @Override
	    public void onPause() {
	        // Add the following line to unregister the Sensor Manager onPause
	        mSensorManager.unregisterListener(mShakeDetector);
	        super.onPause();
	      
	    } 
	
	
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.activity_main, menu);
        return true;
    }

public boolean onOptionsItemSelected(MenuItem item) {
	
	

if(	item.getItemId()==R.id.as)
{

	Intent k=new Intent(rcc.this,about.class);
		
		startActivity(k);
}



else if(item.getItemId()==R.id.t1)
{
	
	if((qux%2)!=0)
	{
		camera=Camera.open();
		Parameters p=camera.getParameters();
		p.setFlashMode(Parameters.FLASH_MODE_TORCH);
		camera.setParameters(p);
		camera.startPreview();
		item.setTitle("LED OFF" );
		
	
	}
	
	else
	{
		camera.stopPreview();
		camera.release();
		item.setTitle("LED ON" );
	}
	qux++;	

}

return true ;

}
}


src/com/example/resistorcolorcodec/MainActivity.java
package com.example.resistorcolorcodec;


//import com.example.resistorcolorcodec.OnSwipeTouchListener.GestureListener;

import java.util.Calendar;

import android.app.Activity;
import android.content.Intent;
import android.hardware.Camera;
import android.hardware.Camera.Parameters;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;


public class MainActivity extends Activity {

	int thud=0;
	MediaPlayer player;
	private Camera camera;
    
	float oldTouchValue;
	@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
       
    Button b1,b2,p;
    b1=(Button) findViewById(R.id.button1);
    b2=(Button) findViewById(R.id.button2);
    p=(Button) findViewById(R.id.p);
    
    final Calendar c= Calendar.getInstance();
    int hr1=16;
    int min=02;
    int s1=c.get(Calendar.HOUR_OF_DAY);
	int s2=c.get(Calendar.MINUTE);
	
	
	if(hr1==s1&&min==s2)
{
		player = MediaPlayer.create(MainActivity.this, R.raw.ed);
		player.start();
		 																					//EASTER EGG PERK CODE
}
    
    

    
    
    
    b1.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View arg0) {
		Intent i=new Intent(MainActivity.this,rcalc.class);
		startActivity(i);
		}
	});
    
 b2.setOnClickListener(new View.OnClickListener() {
	
	@Override
	public void onClick(View arg0) 
	{
	
	 Intent j=new Intent(MainActivity.this,rcc.class);
	 startActivity(j);
		
	}

});
    
 	
   p.setOnClickListener(new View.OnClickListener() {
	
	@Override
	public void onClick(View arg0) 
	{
		thud++;
	
		if(thud>=9)
	{
			thud=0;
			player = MediaPlayer.create(MainActivity.this, R.raw.ed);
			player.start();
		 
	}
	
	}
});
  
    }

	
	

    
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
     getMenuInflater().inflate(R.menu.activity_main, menu);
     menu.findItem(R.id.t1).setEnabled(false);  	//disable LED ON
     menu.findItem(R.id.t1).setVisible(false);		//hide LED ON
     return true;
    
    }
    
    

public boolean onOptionsItemSelected(MenuItem item) {
  
	

if(	item.getItemId()==R.id.as)
{
	Intent k=new Intent(MainActivity.this,about.class);
	startActivity(k);
}

return true ;

}
}


src/com/example/resistorcolorcodec/ShakeDetector.java
package com.example.resistorcolorcodec;



import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.util.FloatMath;
 
public class ShakeDetector implements SensorEventListener {
 
    /*
     * The gForce that is necessary to register as shake.
     * Must be greater than 1G (one earth gravity unit).
     * You can install "G-Force", by Blake La Pierre
     * from the Google Play Store and run it to see how
     *  many G's it takes to register a shake
     */
    private static final float SHAKE_THRESHOLD_GRAVITY = 2.7F;
    private static final int SHAKE_SLOP_TIME_MS = 500;
    private static final int SHAKE_COUNT_RESET_TIME_MS = 3000;
 
    private OnShakeListener mListener;
    private long mShakeTimestamp;
    private int mShakeCount;
 
    public void setOnShakeListener(OnShakeListener listener) {
        this.mListener = listener;
    }
 
    public interface OnShakeListener {
        public void onShake(int count);
    }
 
    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {
        // ignore
    }
 
    @Override
    public void onSensorChanged(SensorEvent event) {
 
        if (mListener != null) {
            float x = event.values[0];
            float y = event.values[1];
            float z = event.values[2];
 
            float gX = x / SensorManager.GRAVITY_EARTH;
            float gY = y / SensorManager.GRAVITY_EARTH;
            float gZ = z / SensorManager.GRAVITY_EARTH;
 
            // gForce will be close to 1 when there is no movement.
            float gForce = FloatMath.sqrt(gX * gX + gY * gY + gZ * gZ);
 
            if (gForce > SHAKE_THRESHOLD_GRAVITY) {
                final long now = System.currentTimeMillis();
                // ignore shake events too close to each other (500ms)
                if (mShakeTimestamp + SHAKE_SLOP_TIME_MS > now) {
                    return;
                }
 
                // reset the shake count after 3 seconds of no shakes
                if (mShakeTimestamp + SHAKE_COUNT_RESET_TIME_MS < now) {
                    mShakeCount = 0;
                }
 
                mShakeTimestamp = now;
                mShakeCount++;
 
                mListener.onShake(mShakeCount);
            }
        }
    }
}

src/com/example/resistorcolorcodec/rcalc.java
package com.example.resistorcolorcodec;


import java.util.ArrayList;
import java.util.List;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.hardware.Camera;
import android.hardware.Camera.Parameters;
import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import com.example.resistorcolorcodec.ShakeDetector.OnShakeListener;


public class rcalc extends Activity {

TextView tv;    
EditText e1;
Spinner s,s2;
CheckBox cb;	
Button ohm,kohm,mohm,c1,c2,c3,t,calc;
long flag;
double x,y,temp,cb1,cb2,cb3,foo;
int flag1;
String o;
boolean torch_engaged;
int qux=1;
double a[]={ 1.0,1.1,1.2,1.3,1.5,1.6,1.8,2.0,2.2,2.4,2.7,3.0,3.3,3.6,3.9,4.3,4.7,5.1,5.6,6.2,6.8,7.5,8.2,9.1,
			10,11,12,13,15,16,18,20,22,24,27,30,33,36,39,43,47,51,56,62,68,75,82,91,
			100,110,120,130,150,160,180,200,220,240,270,300,330,360,390,430,470,510,560,620,680,750,820,910};	
	    

//sensor declarator	
		private SensorManager mSensorManager;
	    private Sensor mAccelerometer;
	    private ShakeDetector mShakeDetector;
	      private Camera camera;
	     
	@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.rcalc);
        tv=(TextView) findViewById(R.id.textView2);
        s = (Spinner) findViewById(R.id.spinner1);
        s2=(Spinner) findViewById(R.id.spinner2);
        cb=(CheckBox) findViewById(R.id.checkBox1);   
        e1=(EditText) findViewById(R.id.editText1);
        c1=(Button) findViewById(R.id.button4);
        c2=(Button) findViewById(R.id.button5);
        c3=(Button) findViewById(R.id.button6);
        calc=(Button) findViewById(R.id.button7);
        t=(Button) findViewById(R.id.button8);
    
        //code for spinner1
    
    List list = new ArrayList();
    list.add("±5%");
    list.add("±10%");
    ArrayAdapter<String> dataAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item,list);
	dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
	s.setAdapter(dataAdapter);
	
	//code for spinner2
	 	List list1 = new ArrayList();
	    list1.add("Ω");
	    list1.add("KΩ");
	    list1.add("MΩ");
	    ArrayAdapter<String> dataAdapter1 = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item,list1);
		dataAdapter1.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
		s2.setAdapter(dataAdapter1);
	
	
	
    
 // ShakeDetector initialisation
    mSensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
    mAccelerometer = mSensorManager
            .getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
    mShakeDetector = new ShakeDetector();
    mShakeDetector.setOnShakeListener(new OnShakeListener() {

        @Override
        public void onShake(int count) {
            /*
             * The following method setup whatever you want done once the
             * device has been shook.
             */
     	 String v="";
     	 e1.setText(v);
     	c1.setBackgroundResource(R.drawable.blank); 
   	 c2.setBackgroundResource(R.drawable.blank);
   	 c3.setBackgroundResource(R.drawable.blank);
   	 t.setBackgroundResource(R.drawable.blank);
        }
    });
    //end sensors
    
    
    //chkbx code
    
    cb.setOnClickListener(new View.OnClickListener() {
		
		@Override
		public void onClick(View v) {
		if(((CheckBox)v).isChecked())
		{
		flag1=1;	
		}
		else
		{
			flag1=0;
		}
		
		}
		});
	
    //end chkbx code
    
    
    
    
    calc.setOnClickListener(new View.OnClickListener() {
		
    	
		@Override
		public void onClick(View arg0) {
			
			// TODO Auto-generated method stub
			   tv.setText("");
			String z=String.valueOf(s.getSelectedItem());
			
			if(z=="±5%")
			    {
			    t.setBackgroundResource(R.drawable.gold);
			    }
			    else if (z=="±10%")    
			    {
			    t.setBackgroundResource(R.drawable.silver);
			    }
				 
String z2=String.valueOf(s2.getSelectedItem());
			

			if(z2=="Ω")
			{
				flag=1;
				o="Ω";
			}
			else if(z2=="KΩ")
			{
				o="KΩ";
				flag=1000;
			}
			else if (z2=="MΩ")
			{
				o="MΩ";
				flag=1000000;
			}
			
		
			//take i/p 
			 try   {
			        x = Double.parseDouble(e1.getText().toString());
			       }
			       catch(Exception e)
			       {
			    	   
			       }
				foo=x;
					 
			 //i/p op finished
	//special case
	if(x>910&&x<=1000)
	 {
	 c1.setBackgroundResource(R.drawable.brown); 
	 c2.setBackgroundResource(R.drawable.black);
	 c3.setBackgroundResource(R.drawable.red);
	 }
	 else if(x>1000)
	 {
		 nonstd();
	 }
			 //
	
	
				if(x>91)
			{
				 cb3=flag*10;
				 y=x/10;
				 cb2= y%10;
				 temp= y-cb2;			
				 cb1=temp/10;
		  
		   if(     cb1==0&&cb2==1||
			       cb1==0&&cb2==2||
  			       cb1==0&&cb2==3||
				   cb1==1&&cb2==0||
				   cb1==1&&cb2==1||
				   cb1==1&&cb2==2||
				   cb1==1&&cb2==3||
				   cb1==1&&cb2==5||
				   cb1==1&&cb2==6||
				   cb1==1&&cb2==8||
				   cb1==2&&cb2==0||
				   cb1==2&&cb2==2||
				   cb1==2&&cb2==4||
				   cb1==2&&cb2==7||
				   cb1==3&&cb2==0||
				   cb1==3&&cb2==3||
				   cb1==3&&cb2==6||
				   cb1==3&&cb2==9||
				   cb1==4&&cb2==3||
				   cb1==4&&cb2==7||
				   cb1==5&&cb2==1||
				   cb1==5&&cb2==6||
				   cb1==6&&cb2==2||
				   cb1==6&&cb2==8||
				   cb1==7&&cb2==5||
				   cb1==8&&cb2==2||
				   cb1==9&&cb2==1
				   )
		   
		   {
		   
		   func(cb1,c1);
		 
		   func(cb2,c2);
			 
		   
		   }
		   else 
		   {
			if(flag1==0)
			{
				nonstd();
			}
			else if(flag1==1)
				
			{

				 for(int ik=0;ik<=a.length;ik++)
				 {
					 if(a[ik]>x)
					 {
						 x=a[ik];
						break; 
					 }
				 }
			
				 cb3=flag*10;
				 y=x/10;
				 cb2= y%10;
				 temp= y-cb2;			
				 cb1=temp/10;
				 tv.setText("Closest std value:"+x+o);
			   
				   func(cb1,c1);
				   func(cb2,c2);
				
			}
		   
		   }
		   }
		

			 
			 
			 
	
		else if (x<=91)
		{
	
			cb3=flag;
			
			if (x!=Math.round(x))
	   {
	  x=x*10;	
	  cb3=flag/10;
	   }
	
			cb2= x%10;
			temp= x-cb2;			
			cb1=temp/10;

  if (    cb1==0&&cb2==1||
		  cb1==0&&cb2==2||
		  cb1==0&&cb2==3||
		  cb1==1&&cb2==0||
		   cb1==1&&cb2==1||
		   cb1==1&&cb2==2||
		   cb1==1&&cb2==3||
		   cb1==1&&cb2==5||
		   cb1==1&&cb2==6||
		   cb1==1&&cb2==8||
		   cb1==2&&cb2==0||
		   cb1==2&&cb2==2||
		   cb1==2&&cb2==4||
		   cb1==2&&cb2==7||
		   cb1==3&&cb2==0||
		   cb1==3&&cb2==3||
		   cb1==3&&cb2==6||
		   cb1==3&&cb2==9||
		   cb1==4&&cb2==3||
		   cb1==4&&cb2==7||
		   cb1==5&&cb2==1||
		   cb1==5&&cb2==6||
		   cb1==6&&cb2==2||
		   cb1==6&&cb2==8||
		   cb1==7&&cb2==5||
		   cb1==8&&cb2==2||
		   cb1==9&&cb2==1
		   )
  
  
  
  
  {
	  if(x==1||x==2||x==3)
	  {
	  beegbug1(x);
	  }
	  else
	  {
	  func(cb1,c1);
	  func(cb2,c2);
	  }	  

  }
  else
  {
	  if(flag1==0)
		{
		  nonstd();
		}
		else if(flag1==1)
			
		{
			 
			double oof = 0;	
			for(int ik=0;ik<=a.length;ik++)
			 {
				 if(a[ik]>foo)
				 {
					 oof=a[ik];
					break; 
				 }
			 }
			  tv.setText("Closest std value:"+oof+o);
			  
		 cb3=flag;
		
     	 if (oof!=Math.round(oof))
		   {
     		oof=oof*10;	
     		cb3=flag/10;
		   }
     	 	cb2= oof%10;
     	 	temp= oof-cb2;			
     	 	cb1=temp/10;
				  
     	 	if(oof==1||oof==2||oof==3)
     		  {
     		  beegbug1(oof);
     		  }
     	 	else
     	 	{
     	 	func(cb1,c1);
     	 	func(cb2,c2);
     	 	}
		}
			  
		}
  }
  
 
		

		
		//code for multiplier
		
		
		if(cb3<1)
		  {
			   c3.setBackgroundResource(R.drawable.gold);
		  }  
		else if(cb3==1)
		  {
			   c3.setBackgroundResource(R.drawable.black);
		  }
		  else if(cb3==10)
		  {
			   c3.setBackgroundResource(R.drawable.brown);
		  }
		  else if(cb3==100)
		  {
			   c3.setBackgroundResource(R.drawable.red);
		  }
		  else if(cb3==1000)
		  {
			   c3.setBackgroundResource(R.drawable.orange);
		  }
		  else if(cb3==10000)
		  {
			   c3.setBackgroundResource(R.drawable.yellow);
		  }
		  else if(cb3==100000)
		  {
			   c3.setBackgroundResource(R.drawable.green);
		  }
		  else if(cb3==1000000)
		  {
			   c3.setBackgroundResource(R.drawable.blue);
		  }
		  else if(cb3==10000000)
		  {
			   c3.setBackgroundResource(R.drawable.violet);
		  }
		  else if(cb3==100000000)
		  {
			   c3.setBackgroundResource(R.drawable.grey);
		  }
		  else if(cb3==1000000000)
		  {
			   c3.setBackgroundResource(R.drawable.white);
		  }
		
		  else if(cb3==221)
		  {
			   c3.setBackgroundResource(R.drawable.blank);
			   c2.setBackgroundResource(R.drawable.blank);
			   c1.setBackgroundResource(R.drawable.blank);
				  
		  }
		
		}
	});


	}
    
	 @Override
	    public void onResume() {
	        super.onResume();
	        // Add the following line to register the Session Manager Listener onResume
	        mSensorManager.registerListener(mShakeDetector, mAccelerometer,    SensorManager.SENSOR_DELAY_UI);
	    }
	 
	    @Override
	    public void onPause() {
	        // Add the following line to unregister the Sensor Manager onPause
	        mSensorManager.unregisterListener(mShakeDetector);
	        super.onPause();
	      
	    } 
	    
	    
	    //nonstd defn
	public void nonstd()
	{
		Context context = getApplicationContext();
   		CharSequence text="Enter standard value ";
   		int duration = Toast.LENGTH_SHORT;
		Toast toast = Toast.makeText(context, text, duration);
		toast.show();
		cb3=221;
		t.setBackgroundResource(R.drawable.blank);
	}
	//func defn
	public void func(double z,Button b)
	{
		 if(z==0)
		   {
			   b.setBackgroundResource(R.drawable.black);
		   }
		   else if(z==1)
		   {
			   b.setBackgroundResource(R.drawable.brown);
		   }
		   else if(z==2)
		   {
			   b.setBackgroundResource(R.drawable.red);
		   }
		   else if(z==3)
		   {
			   b.setBackgroundResource(R.drawable.orange);
		   }
		   else if(z==4)
		   {
			   b.setBackgroundResource(R.drawable.yellow);
		   }
		   else if(z==5)
		   {
			   b.setBackgroundResource(R.drawable.green);
		   }
		   else if(z==6)
		   {
			   b.setBackgroundResource(R.drawable.blue);
		   }
		   else if(z==7)
		   {
			   b.setBackgroundResource(R.drawable.violet);
		   }
		   else if(z==8)
		   {
			   b.setBackgroundResource(R.drawable.grey);
		   }
		   else if(z==9)
		   {
			   b.setBackgroundResource(R.drawable.white);
		   }
		   	
	}
	
	
	public void beegbug1(double z)
	{
		if(z==1)
		{
			c1.setBackgroundResource(R.drawable.brown); 
		   	 c2.setBackgroundResource(R.drawable.black);
		   	cb3=flag/10;
		   	 
		}
		else if(z==2)
		{

			c1.setBackgroundResource(R.drawable.red); 
		   	 c2.setBackgroundResource(R.drawable.black);
		   	cb3=flag/10;
		}
		else if(z==3)
		{

			c1.setBackgroundResource(R.drawable.orange); 
		   	 c2.setBackgroundResource(R.drawable.black);
		   	cb3=flag/10;
		}
	}
	
	
@Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.activity_main, menu);
        
return true;

}
 


public boolean onOptionsItemSelected(MenuItem item) {
	
	

if(	item.getItemId()==R.id.as)
{

	Intent k=new Intent(rcalc.this,about.class);
		
		startActivity(k);
}



else if(item.getItemId()==R.id.t1)
{
	torch_engaged=true;
	if((qux%2)!=0)
	{
		
		camera=Camera.open();
		Parameters p=camera.getParameters();
		p.setFlashMode(Parameters.FLASH_MODE_TORCH);
		camera.setParameters(p);
		camera.startPreview();
		item.setTitle("LED OFF" );
		
	
	}
	
	else
	{
		camera.stopPreview();
		camera.release();
		item.setTitle("LED ON" );
		
		
	}
	qux++;	

}

return true ;

}
}
